command/agent/sink/sink.go

```diff
@@ -71,8 +71,30 @@ func NewSinkServer(conf *SinkServerConfig) *SinkServer {
 // Run executes the server's run loop, which is responsible for reading
 // in new tokens and pushing them out to the various sinks.
 func (ss *SinkServer) Run(ctx context.Context, incoming chan string, sinks []*SinkConfig) {
+	latestToken := new(string)
```

Comment:
What's the benefit of all the rearranging in this file?

Reply:
I feel it improves readability.  The function that was being sent over a channel doesn't need to close over anything except latestToken, and we don't need to create a new func each iteration.  Instead just pass the relevant items, namely the token and sink to work on.

When I was trying to understand what the function was doing, the fact that a closure was being created and sent over the channel made it seem like there was more going on than there was, and understanding the nature of the communication required that I read and make sense of the function and what it's doing with latestToken.  In general I think it's best to define closures in the most narrow scope possible as it allows the reader to more quickly judge how it interacts with the surrounding code/state.